### ISSUE TITLE
Bus error, exit code 135, when exponentiating Vector with other unequal sized Vector

### ISSUE THREAD

**Issue by azadad96** [JuliaLang/julia#45401]
### Explanation:
When trying to take one `Vector` of size `n` to the power of another `Vector` of size `m` using the `.^` operator, `Bus error (core dumped)` is printed, after which julia exits with the error code `135`.
It is to be noted that `n` and `m` must both be larger than 1 and different from each other.

### Expected Behaviour:
Some error message should be printed (at least in the REPL) and julia shouldn't exit.

### Steps to reproduce:
run `[1, 2] .^ [3, 4, 5]`

### versioninfo:
```
Julia Version 1.6.3
Commit ae8452a9e0* (2021-09-23 17:34 UTC)
Platform Info:
  OS: Linux (x86_64-suse-linux)
  CPU: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
```

**Comment by oscardssmith** [comment]
How did you install Julia? I can't reproduce this on 1.6.5 on Ubuntu.

**Comment by azadad96** [comment]
> How did you install Julia? I can't reproduce this on 1.6.5 on Ubuntu.

I installed it using the zypper package manager on openSUSE tumbleweed from the Main Repository.
I'm currently also compiling julia myself to test it there.

**Comment by azadad96** [comment]
Hmm, I've tested it on version 1.7.3 now and it seems to be working.

```
julia> [1, 2] .^ [3, 4, 5]
ERROR: DimensionMismatch("arrays could not be broadcast to a common size; got a dimension with lengths 2 and 3")
Stacktrace:
 [1] _bcs1
   @ ./broadcast.jl:516 [inlined]
 [2] _bcs
   @ ./broadcast.jl:510 [inlined]
 [3] broadcast_shape
   @ ./broadcast.jl:504 [inlined]
 [4] combine_axes
   @ ./broadcast.jl:499 [inlined]
 [5] instantiate
   @ ./broadcast.jl:281 [inlined]
 [6] materialize(bc::Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{1}, Nothing, typeof(^), Tuple{Vector{Int64}, Vector{Int64}}})
   @ Base.Broadcast ./broadcast.jl:860
 [7] top-level scope
   @ REPL[1]:1
```
Seems to have been fixed since v1.6.3, I'll close this issue for now

**Comment by oscardssmith** [comment]
Unfortunately most OSes that package Julia break it by forcing it to use broken versions of LLVM.
